tools/content/unimplemented_features_test.dart

```diff
@@ -0,0 +1,147 @@
+// Override `fluter test`'s default timeout
+@Timeout(Duration(minutes: 10))
+library;
+
+import 'dart:io';
+import 'dart:math';
+
+import 'package:checks/checks.dart';
+import 'package:html/dom.dart' as dom;
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/model/content.dart';
+
+import 'model.dart';
+
+
+/// Check if there are unimplemented features from the given corpuses of HTML
+/// contents from Zulip messages.
+///
+/// This test is meant to be run via `tools/content/check-features`.
+///
+/// To run it directly, use:
+///
+///   flutter test tools/content --dart-define=corpusDir=path/to/corpusDir
+///
+/// where `path/to/corpusDir` should be a directory containing files with
+/// outputs generated from tools/content/fetch_messages.dart.
+///
+/// Optionally, you can enable more details with `--dart-define=verbose=true`.
+///
+/// The test writes an overview of unimplemented features at the beginning to
+/// standard output, followed by the details of each feature.  To look for live
+/// examples, you can search on the Zulip community by message ID from all
+/// public channels.
+///
+/// For example, a search query like "near: 12345 channels: public" would work.
+///
+/// See also:
+/// * lib/model/content.dart, which implements of the content parser.
+/// * tools/content/fetch_messages.dart, which produces the corpuses.
+void main() async {
```

Comment:
The test can be run manually via:
        `flutter test --dart-define=corpusDir=path/to/corpusDir tools/content`

This is a pretty funky command line.

There's a good reason it's done this way, and we smooth it over in the end by having a nice wrapper script. But the wrapper script should appear in the same commit as this file does, so that the "test" file offers a reasonable way to run it from the beginning.

If you want to split the changes into two commits, probably a good way to do so would be one commit for fetching and another for parsing.

Reply:
Putting them in the same commit makes sense. There is not much complexity that can be resolved by a sequence of separate ones.